lib/Sema/CSDiag.cpp

```diff
@@ -684,18 +684,39 @@ static void fixItChangeInoutArgType(const Expr * arg,
   SmallString<32> scratch;
   SourceLoc endLoc;       // Filled in if we decide to diagnose this
   SourceLoc startLoc;     // Left invalid if we're inserting
-  
-  auto isSimpleTypelessPattern = [](Pattern *P) -> bool {
+
+  /// Does this declaration share its type annotation with the next one,
+  /// like "x" does in "let x, y: Int"?
+  auto isSharedWithNextVar = [](SourceRange typeRange, VarDecl *VD) {
+    auto *PBD = VD->getParentPatternBinding();
+    // If there's no PBD, there's no type name sharing.
+    if (!PBD) return false;
+    
+    auto nextIndex = PBD->getPatternEntryIndexForVarDecl(VD) + 1;
+    // If VD is last, there's nothing we could be sharing a type with.
```

Comment:
…What about the previous one? The fix-it you have in the example will change the type of both variables. Again, I don't think that's unreasonable, but in that case it should be the same in either direction.

Reply:
There's a reason I reverted most of this commit and kept only the structure of the test. :^)

Reply:
I was planning to wait and squash-merge this, but I think I'll just squash it immediately to simplify the history.

Reply:
Yep, sorry, GitHub showed me the in-between state and immediately after I saw that you dropped it.